homeassistant/components/romy/config_flow.py

```diff
@@ -0,0 +1,151 @@
+            self.romy_info[CONF_HOST] = user_input[CONF_HOST]
+
+            new_romy = await romy.create_romy(self.romy_info[CONF_HOST], "")
+
+            if not new_romy.is_initialized:
+                errors[CONF_HOST] = "cannot_connect"
+            else:
+                # check if already setuped
+                await self.async_set_unique_id(new_romy.unique_id)
+                self._abort_if_unique_id_configured()
+
+                self.romy_info[CONF_NAME] = new_romy.name
+
+                if not new_romy.is_unlocked:
+                    return await self.async_step_unlock_http_interface()
+
+            if not errors:
+                return await self._async_step_finish_config()
+
+        return self.async_show_form(step_id="user", data_schema=data, errors=errors)
+
+    async def async_step_unlock_http_interface(
+        self, user_input: dict[str, str] | None = None
+    ) -> FlowResult:
+        """Unlock the http interface with password if robot is locked."""
+        errors: dict[str, str] = {}
+        data = _schema_with_password()
+
+        if user_input:
+            self.romy_info[CONF_PASSWORD] = user_input[CONF_PASSWORD]
+            new_romy = await romy.create_romy(
+                self.romy_info[CONF_HOST], self.romy_info[CONF_PASSWORD]
+            )
+
+            if not new_romy.is_initialized or not new_romy.is_unlocked:
+                errors[CONF_PASSWORD] = "invalid_auth"
+
+            if not errors:
+                return await self._async_step_finish_config()
+
+        return self.async_show_form(
+            step_id="unlock_http_interface", data_schema=data, errors=errors
+        )
+
+    async def async_step_zeroconf(
+        self, discovery_info: zeroconf.ZeroconfServiceInfo
+    ) -> FlowResult:
+        """Handle zeroconf discovery."""
+
+        LOGGER.debug("Zeroconf discovery_info: %s", discovery_info)
+
+        # connect and gather information from your ROMY
+        host = discovery_info.host
+        LOGGER.debug("ZeroConf Host: %s", host)
+
+        new_discovered_romy = await romy.create_romy(host, "")
+
+        name = new_discovered_romy.name
+        LOGGER.debug("ZeroConf Name: %s", name)
+
+        # get unique id and stop discovery if robot is already added
+        unique_id = new_discovered_romy.unique_id
+        LOGGER.debug("ZeroConf Unique_id: %s", unique_id)
+        await self.async_set_unique_id(unique_id)
+        self._abort_if_unique_id_configured()
+
+        self.context.update(
+            {
+                "title_placeholders": {"name": f"{name} ({host} / {unique_id})"},
+                "configuration_url": f"http://{host}:{new_discovered_romy.port}",
+            }
+        )
+
+        self.romy_info[CONF_HOST] = host
+        self.romy_info[CONF_NAME] = name
+        self.romy_info[CONF_UUID] = unique_id
+
+        # if robot is already unlocked add it directly
+        if new_discovered_romy.is_initialized and new_discovered_romy.is_unlocked:
```

Comment:
If `new_discovered_romy.is_initialized` is false then it seems like moving to the password step may not be quite right.  Should it abourt/fail to be discovered and be added manually? Not sure how other integrations handle this.

Reply:
saw that too but didn't care/handle it cause its a real small corner case imho, roboter must exaclty went offline in this moment

go back to step user in case of romy uninitialized error